tests/e2e/clis/clis_test.go

```
@@ -722,6 +745,69 @@ type cellResult struct {
+//
+// Only the harness's own artefacts are deleted, rather than os.RemoveAll on the
+// directory: anything else a developer has parked in reports/ survives, and a
+// caller who points reportsDir somewhere unwise cannot lose unrelated data. A
+// missing directory is success, not an error.
+func clearReportsDir(dir string) error {
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		return nil
+	}
+	return filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil || d.IsDir() {
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 797) but it returns nil

(nilerr)